### DISCUSSION TITLE
Evaluation - Precision/Recall/F1-score

### DISCUSSION THREAD

**Discussion by bukharidxb** [lmcinnes/pynndescent#153] [General]
Please let me know how can I evaluate it with other ML techniques like precision, recall, f1-score etc. Also, if possible can I get the extract of the confusion matrix. My dataset has 11 classes (labels).

Thank you

**Comment by lmcinnes** [comment]
This library just computes which elements are near neighbors. If you want to actually use it for classification or regression pipelines you'll need to hook this up with sklearn's KNeighborClassifier or similar. This is not so hard to do -- see the [docs](https://pynndescent.readthedocs.io/en/latest/pynndescent_in_pipelines.html) to get started, and also issue #146

**Reply by bukharidxb** [reply]
First of all, thank you so much for your prompt reply. Please ignore my stupid questions as I am a technical guy in the network area in computer science and have a little bit of idea on how to use python. I used your code for my dataset and got an accuracy of 98%, does this means once I use the Pynndescent for the classification, I will get the same accuracy overall?

My next question is, I have already gone through the docs prior to writing comments here. As suggested in issue # 146, I used the code for my X_train, y_train, X_test and y_test partition of the dataset but unfortunately got some error,

here is my code,

X_train, X_test, y_train, y_test = train_test_split(X_train_reshape, Y, test_size=0.25,random_state=42)

fast_classifier = make_pipeline(
    PyNNDescentTransformer(),
    KNeighborsClassifier(metric='precomputed')

fast_classifier.fit_transform(X_train, y_train)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-17-654f7506b69d> in <module>
----> 1 fast_classifier.transform(X_train,y_train)

~\anaconda3\lib\site-packages\sklearn\utils\metaestimators.py in __get__(self, obj, owner)
    108             # this is to allow access to the docstrings.
    109             if not self.check(obj):
--> 110                 raise attr_err
    111 
    112             # lambda, but not partial, allows help() to work with update_wrapper

AttributeError: This 'Pipeline' has no attribute 'transform'


Please help so I can use this library for my project and compare the result with other classifiers of ML.

Thank you
